### DISCUSSION TITLE
Is there a way to enable the partial text selection for copy and paste?

### DISCUSSION THREAD

**Discussion by lan680** [danielsaidi/RichTextKit#191] [Q&A]
Love RichTextKit! It has been working greatly.
 
Just trying to add the partial text selection function, but we are running to the compatibility issue - the rich text string becomes inconsistent due to compatibility issues between `UITextView` and SwiftUI, and the `UITextView` exhibits unusual behavior with respect to its width.
 
The code used is the following -
struct SelectableTextView: UIViewRepresentable {
    let text: String
   
    func makeUIView(context: Context) -> UITextView {
        let textView = UITextView()
        textView.isEditable = false
        textView.isSelectable = true
        textView.text = text
        textView.font = UIFont.systemFont(ofSize: 17)
        return textView
    }
   
    func updateUIView(_ uiView: UITextView, context: Context) {
        uiView.text = text
    }
}
 
Any suggestion or guidance will be greatly appreciated!

**Comment by danielsaidi** [comment]
Hi @lan680 

I'm happy to hear that you like it! ...but I'm not sure how this relates to RichTextKit? 🤔

**Reply by lan680** [reply]
Hi Daniel, I'm a new learner to SwiftUI. 

I've opened a issue post along with a screenshot at https://www.reddit.com/r/SwiftUI/comments/1fts4uw/how_to_select_partial_text_using_swiftui/

If you get a chance, would you please take a look? Would love to hear your suggestion. Thank you!
